pandas/io/tests/test_sql.py

```diff
@@ -2425,44 +2308,68 @@ def clean_up(test_table_to_drop):
                           frame=df_if_exists_1,
                           con=self.conn,
                           name=table_name,
-                          flavor='sqlite',
                           if_exists='notvalidvalue')
         clean_up(table_name)
 
         # test if_exists='fail'
-        sql.to_sql(frame=df_if_exists_1, con=self.conn, name=table_name,
-                   flavor='sqlite', if_exists='fail')
+        sql.to_sql(frame=df_if_exists_1, con=self.conn,
+                   name=table_name, if_exists='fail')
         self.assertRaises(ValueError,
                           sql.to_sql,
                           frame=df_if_exists_1,
                           con=self.conn,
                           name=table_name,
-                          flavor='sqlite',
                           if_exists='fail')
 
         # test if_exists='replace'
         sql.to_sql(frame=df_if_exists_1, con=self.conn, name=table_name,
-                   flavor='sqlite', if_exists='replace', index=False)
+                   if_exists='replace', index=False)
         self.assertEqual(sql.tquery(sql_select, con=self.conn),
                          [(1, 'A'), (2, 'B')])
         sql.to_sql(frame=df_if_exists_2, con=self.conn, name=table_name,
-                   flavor='sqlite', if_exists='replace', index=False)
+                   if_exists='replace', index=False)
         self.assertEqual(sql.tquery(sql_select, con=self.conn),
                          [(3, 'C'), (4, 'D'), (5, 'E')])
         clean_up(table_name)
 
         # test if_exists='append'
         sql.to_sql(frame=df_if_exists_1, con=self.conn, name=table_name,
-                   flavor='sqlite', if_exists='fail', index=False)
+                   if_exists='fail', index=False)
         self.assertEqual(sql.tquery(sql_select, con=self.conn),
                          [(1, 'A'), (2, 'B')])
         sql.to_sql(frame=df_if_exists_2, con=self.conn, name=table_name,
-                   flavor='sqlite', if_exists='append', index=False)
+                   if_exists='append', index=False)
         self.assertEqual(sql.tquery(sql_select, con=self.conn),
                          [(1, 'A'), (2, 'B'), (3, 'C'), (4, 'D'), (5, 'E')])
         clean_up(table_name)
 
 
+class TestSQLFlavorDeprecation(tm.TestCase):
+    """
+    gh-13611: test that the 'flavor' parameter
+    is appropriately deprecated by checking the
+    functions that directly raise the warning
+    """
+
+    con = 1234  # don't need real connection for this
+    funcs = ['SQLiteDatabase', 'pandasSQL_builder']
```

Comment:
to_sql isn't just the only function that calls pandasSQL_builder though that I changed - seems more reasonable to test the function underlying all the ones whose signatures I changed IMO.

Also, why is pandasSQL_builder not need?  It has no underscore prefix --> not an internal function technically.